src/overlays/actors/ovl_En_Elfgrp/z_en_elfgrp.c

```diff
@@ -61,113 +100,118 @@ void func_80A396B0(EnElfgrp* this, s32 numCutscenes) {
-                func_80A396B0(this, 2);
-                return;
+                EnElfgrp_SetCutscene(this, 2);
+                break;
             }
 
-            if ((func_80A39BD0(play, this->unk_147) + sp24) >= 25) {
+            if ((EnElfgrp_GetHeldFairiesCount(play, this->type) + numberInFountain) >= STRAY_FAIRY_TOTAL) {
                 this->actionFunc = func_80A3A398;
 
-                switch (this->unk_147) {
-                    case ENELFGRP_1:
-                        if (CHECK_WEEKEVENTREG(WEEKEVENTREG_23_02)) {
-                            func_80A396B0(this, 1);
+                switch (this->type) {
+                    case ENELFGRP_TYPE_POWER:
+                        if (CHECK_WEEKEVENTREG(WEEKEVENTREG_OBTAINED_GREAT_SPIN_ATTACK)) {
+                            EnElfgrp_SetCutscene(this, 1);
                         } else {
-                            this->unk_14A |= 4;
+                            this->stateFlags |= ELFGRP_STATE_2;
                         }
                         break;
 
-                    case ENELFGRP_2:
+                    case ENELFGRP_TYPE_WISDOM:
                         if (gSaveContext.save.saveInfo.playerData.isDoubleMagicAcquired == true) {
-                            func_80A396B0(this, 1);
+                            EnElfgrp_SetCutscene(this, 1);
                         }
                         break;
 
-                    case ENELFGRP_3:
+                    case ENELFGRP_TYPE_COURAGE:
                         if (gSaveContext.save.saveInfo.playerData.doubleDefense) {
-                            func_80A396B0(this, 1);
+                            EnElfgrp_SetCutscene(this, 1);
                         }
                         break;
 
-                    case ENELFGRP_4:
+                    case ENELFGRP_TYPE_KINDNESS:
                         if (INV_CONTENT(ITEM_SWORD_GREAT_FAIRY) == ITEM_SWORD_GREAT_FAIRY) {
-                            func_80A396B0(this, 1);
+                            EnElfgrp_SetCutscene(this, 1);
                         } else {
-                            this->unk_14A |= 0x10;
+                            this->stateFlags |= ELFGRP_STATE_4;
                         }
                         break;
 
                     default:
                         break;
                 }
-            } else if (func_80A39BD0(play, this->unk_147)) {
+            } else if (EnElfgrp_GetHeldFairiesCount(play, this->type)) {
                 this->actionFunc = func_80A3A7FC;
-                this->actor.textId = (this->unk_147 * 3) + 0x581;
+                this->actor.textId = (this->type * 3) + 0x581;
             } else {
                 this->actionFunc = func_80A3A8F8;
-                if ((gSaveContext.save.saveInfo.weekEventReg[9] & this->unk_146)) {
-                    this->actor.textId = (this->unk_147 * 3) + 0x580;
+
+                if ((gSaveContext.save.saveInfo.weekEventReg[9] & this->talkedOnceFlag)) { // talked for first time
```

Comment:
Should we make macros for this? At the least could probably add comments in z64save.h.

Reply:
We should definitely made macros for this, the top of the file has a comment stating it lol.
I'll add an extra comment in z64save.h in the meanwhile